tests/integration/tasks/test_databricks.py

```diff
@@ -305,35 +305,40 @@ def test_upsert():
             "ContactEMail": "Alexander.Chambers@KentLyons.com",
             "MailingCity": "New Nicholaschester",
         },
+        {
+            "Id": "UpsertTest2",
+            "AccountId": "Updated!EHNYKjSZsiy",
+            "Name": "new upsert-2",
+            "FirstName": "Updated",
+            "LastName": "Carter2",
+            "ContactEMail": "Adam.Carter@TurnerBlack.com",
+            "MailingCity": "Updated!Jamesport",
+        },
     ]
-    upsert_data = [
+
+    data = [
         {
-            "Id": "wRACnHTeuw",
-            "AccountId": "Upserted!",
-            "Name": "Upsert-Merritt",
-            "FirstName": "Melody",
-            "LastName": "Cook",
-            "ContactEMail": "Melody.Cook@ScottMerritt.com",
-            "MailingCity": "Elizabethfurt",
+            "Id": "UpsertTest2",
+            "AccountId": "Updated!EHNYKjSZsiy",
+            "Name": "new upsert-2",
+            "FirstName": "Updated",
+            "LastName": "Carter2",
+            "ContactEMail": "Adam.Carter@TurnerBlack.com",
+            "MailingCity": "Updated!Jamesport",
         }
     ]
+
     table_name = "raw.test_table"
 
-    match = {
-        "Id": "main.Id",
-        "AccountId": "updates.AccountId",
-        "Name": "updates.Name",
-        "FirstName": "updates.FirstName",
-        "LastName": "updates.LastName",
-        "ContactEMail": "updates.ContactEMail",
-        "MailingCity": "main.MailingCity",
-    }
+    match = "Id= main.Id, AccountId= updates.AccountId, Name= updates.Name, FirstName= updates.FirstName, LastName= updates.LastName, ContactEMail= updates.ContactEMail, MailingCity= main.MailingCity"
```

Comment:
1. Noted and fixed

2. Previous implementations used dictionaries, and the final result needs to be a string. Both got rejected (first because the dictionaries were converted to a string, second because it was different from `utils.build_merge_query()`) and as such, I opted to using a string directly.

3. `utils.build_merge_query()` assumes that the user wants to update all the fields in every single upsert operation; it does not give them the option of selectively updating certain fields while allowing other fields to retain their old values. This function allows them to specify which fields will be updated, and which ones will be left intact. For example:
` match="Id=main.Id, AccountId=updates.AccountId, Name=updates.Name, FirstName=updates.FirstName, LastName=updates.LastName, ContactEMail=updates.ContactEMail, MailingCity=main.MailingCity"`
Here, we specify that we want to insert the value of `MailingCity` corresponding to the `main` table, which is the existing/main table that we read from Databricks; in other words, this field won't be updated. The other columns (besides the primary key, but that's irrelevant to the end result) have their values updated using the values from the `updates` temporary view, which is the staging table.

4. For a. the reason described in point 3 and b. The query used to retrieve the column names in `utils.build_merge_query()` does not work with Databricks/SparkSQL; it requires a different SQL statement

5. Noted and fixed